src/lazy/expanded/sequence.rs

```diff
@@ -1,16 +1,70 @@
+use bumpalo::collections::Vec as BumpVec;
+
+use crate::element::iterators::SymbolsIterator;
 use crate::lazy::decoder::{LazyDecoder, LazyRawSequence, LazyRawValueExpr, RawValueExpr};
-use crate::lazy::expanded::macro_evaluator::{TransientEExpEvaluator, TransientTdlMacroEvaluator};
-use crate::lazy::expanded::template::TemplateSequenceIterator;
+use crate::lazy::expanded::macro_evaluator::{MacroEvaluator, RawEExpression, ValueExpr};
+use crate::lazy::expanded::template::{
+    AnnotationsRange, ExprRange, TemplateMacroRef, TemplateSequenceIterator,
+};
 use crate::lazy::expanded::{
     EncodingContext, ExpandedAnnotationsIterator, ExpandedAnnotationsSource, ExpandedValueSource,
     LazyExpandedValue,
 };
-use crate::{Annotations, IonResult, IonType, Sequence};
+use crate::result::IonFailure;
+use crate::{IonError, IonResult, IonType};
+
+#[derive(Copy, Clone, Debug)]
+pub struct Environment<'top, 'data, D: LazyDecoder<'data>> {
```

Comment:
🗺️ The series of (not yet evaluated) expressions passed to a macro as arguments.

`TODO` for me: doc comment